src/main/java/step2/Calculator.java

```diff
@@ -0,0 +1,68 @@
+package step2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Calculator {
+
+    public int add(String text) {
+        if (text == null) {
+            return 0;
+        }
+
+        if (text.isEmpty()) {
+            return 0;
+        }
+
+        if (text.startsWith("//")) {
+            return parseWithCustomDelimiter(text);
+        }
+
+        return parseWithDefaultDelimiter(text);
+    }
+
+    private int parseWithDefaultDelimiter(String text) {
+        String regex = ",|:";
+        String[] strings = text.split(regex);
+        return calculateSum(strings);
+    }
+
+    private int parseWithCustomDelimiter(String text) {
+        Pattern pattern = Pattern.compile("//(.)\n(.*)");
+        Matcher matcher = pattern.matcher(text);
+
+        if (matcher.find()) {
+            String customDelimiter = matcher.group(1);
+            String[] strings = matcher.group(2).split(customDelimiter);
+
+            return calculateSum(strings);
+        }
+
+        return 0;
```

Comment:
```suggestion
        if (!matcher.find()) {
            return 0;
        }
        
        String customDelimiter = matcher.group(1);
        String[] strings = matcher.group(2).split(customDelimiter);
        return calculateSum(strings);
```

위와 같이 구현하는 것은 어떻게 생각하나?

Reply:
if문 안의 로직이 길어질 때, 해당 방법이 좋을 것 같습니다.